cg/store/database.py

```diff
@@ -0,0 +1,44 @@
+from typing import Optional
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import Session, scoped_session, sessionmaker
+
+from cg.exc import CgError
+from cg.store.models import Model
+
+SESSION: Optional[Session] = None
+ENGINE = None
+
+
+def initialise_database(db_uri: str):
+    global SESSION, ENGINE
+    ENGINE = create_engine(db_uri, pool_pre_ping=True)
+    session_factory = sessionmaker(ENGINE)
+    SESSION = scoped_session(session_factory)
+
+
+def get_session() -> Session:
+    if not SESSION:
+        raise CgError("Database not initialised")
+    return SESSION()
+
+
+def get_engine():
+    if not ENGINE:
+        raise CgError("Database not initialised")
+    return ENGINE
+
+
+def create_all_tables() -> None:
+    session: Session = get_session()
+    Model.metadata.create_all(bind=session.get_bind())
```

Comment:
just because I am unfamiliar with session handling. This take the global variable from `initialise_database` can it happen that global variable is overwritten unexpectedly? My question would it be better to be more explicit and pass a session here, or does it not matter?

Reply:
I don't think it matters in this case, but we will likely need to make a choice later for other methods using the session whether we want to pass it as a parameter or just retrieve a global `scoped_session`